bindings/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,175 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# This software and the related documents are Intel copyrighted materials,
+# and your use of them is governed by the express license under which they
+# were provided to you ("License"). Unless the License provides otherwise,
+# you may not use, modify, copy, publish, distribute, disclose or transmit
+# this software or the related documents without Intel's prior written
+# permission.
+#
+# This software and the related documents are provided as is, with no
+# express or implied warranties, other than those that are expressly stated
+# in the License.
+
+cmake_minimum_required(VERSION 3.21)
+project(svs_runtime VERSION 0.0.10 LANGUAGES CXX)
+set(TARGET_NAME svs_runtime)
+
+set(SVS_RUNTIME_HEADERS
+    include/IndexSVSImplDefs.h
+    include/IndexSVSFlatImpl.h
+    include/IndexSVSVamanaImpl.h
+)
+
+set(SVS_RUNTIME_SOURCES
+    src/IndexSVSImplUtils.h
+    src/IndexSVSFlatImpl.cpp
+    src/IndexSVSVamanaImpl.cpp
+)
+
+option(SVS_RUNTIME_ENABLE_LVQ "Enable compilation of SVS runtime with LVQ support" ON)
+if (SVS_RUNTIME_ENABLE_LVQ)
+    message(STATUS "SVS runtime will be built with LVQ support")
+    list(APPEND SVS_RUNTIME_HEADERS
+        include/IndexSVSVamanaLVQImpl.h
+    )
+    list(APPEND SVS_RUNTIME_SOURCES
+        src/IndexSVSVamanaLVQImpl.cpp
+    )
+else()
+    message(STATUS "SVS runtime will be built without LVQ support")
+endif()
+
+option(SVS_RUNTIME_ENABLE_LEANVEC "Enable compilation of SVS runtime with LeanVec support" ON)
+if (SVS_RUNTIME_ENABLE_LEANVEC)
```

Comment:
I'd say SVS_RUNTIME_ENABLE_LVQ and SVS_RUNTIME_ENABLE_LEANVEC can be combined - users either have both or neither